### PR TITLE
Changes to make it "work" on macOS

### PR DESCRIPTION
**Pull request by vespakoen**
Not great, but a start ;)

Looks like some C++20 features are not implemented in Apple Clang, perhaps we can try to use LLVM or GCC from homebrew, but for now, we have bigger problems on macOS with OpenGL (see below).

Not sure what is the best way forward, and if we should even support macOS since OpenGL is deprecated on it and support could go away anytime, but it might be worth it for the time being...

Note: I am not a C++ developer.

Anyways, the application in general is working better than expected, I was only able to draw some lines, but this might be because I don't understand the program yet, most likely because I commented out some GL calls ;)

### CODE REVIEW

**Comment by vespakoen** [comment]
I tried to open the sample file and was amazed by how well it works

Screenshot 2023-11-15 at 01 13 26

Selecting points / lines doesn't seem to work on macOS at the moment.

**Comment by vespakoen** [comment]
Got a bit further with things, the issue turned out to be the `samples` in calls to `glRenderbufferStorageMultisample`, on macOS it only seems to accept `1` as it's value.

Another issue I am running into is with meson's framework handling coming from cmake (reported it over here: https://github.com/mesonbuild/meson/issues/12622)

So right now at the linking step I copy it's output, replace all the `/Full/Paths/To/FrameworkName.framework` with `-framework FrameworkName` and execute it manually in the build folder.

Now I am able to draw and select things and in general things seem to be working ok-ish.

Unfortunately `glLineWidth` with a value above 1 is not working on macOS (and actually is deprecated I think) so turning those lines into quads would be really great ;)

**Comment by tkornack** [comment]
Do you have a more explicit recipe for building on macOS at this point? In my case cmake couldn't find gtk4; though all I tried was installing `gtk4` via homebrew.

**Comment by vespakoen** [comment]
You can try with these (might have missed one or two things ;)):

```
brew install meson eigen opencascade pkg-config gtk4 gtkmm4 glm range-v3 mimalloc pygobject3
```

Here is my build.sh script (note that the paths will probably be different on your machine.

```
#!/usr/bin/env bash

meson setup build

meson compile -C build

cd build

c++  -o dune3d dune3d.p/meson-generated_.._version_gen.cpp.o dune3d.p/meson-generated_.._resources.c.o dune3d.p/meson-generated_.._icon_texture_map.cpp.o dune3d.p/meson-generated_.._color_presets.cpp.o dune3d.p/src_main.cpp.o dune3d.p/src_dune3d_application.cpp.o dune3d.p/src_dune3d_appwindow.cpp.o dune3d.p/src_editor_editor.cpp.o dune3d.p/src_editor_editor_workspace_browser.cpp.o dune3d.p/src_canvas_canvas.cpp.o dune3d.p/src_canvas_gl_util.cpp.o dune3d.p/src_canvas_base_renderer.cpp.o dune3d.p/src_canvas_background_renderer.cpp.o dune3d.p/src_canvas_face_renderer.cpp.o dune3d.p/src_canvas_point_renderer.cpp.o dune3d.p/src_canvas_line_renderer.cpp.o dune3d.p/src_canvas_glyph_renderer.cpp.o dune3d.p/src_canvas_glyph_3d_renderer.cpp.o dune3d.p/src_canvas_box_selection.cpp.o dune3d.p/src_canvas_bitmap_font_util.cpp.o dune3d.p/src_canvas_bitmap_font_bitmap_font_desc.c.o dune3d.p/src_canvas_bitmap_font_bitmap_font_img.c.o dune3d.p/src_canvas_icon_renderer.cpp.o dune3d.p/src_canvas_appearance.cpp.o dune3d.p/src_canvas_selectable_ref.cpp.o dune3d.p/src_import_step_step_importer.cpp.o dune3d.p/src_import_step_step_import_manager.cpp.o dune3d.p/src_util_uuid.cpp.o dune3d.p/src_document_document.cpp.o dune3d.p/src_document_entity_entity.cpp.o dune3d.p/src_document_entity_entity_and_point.cpp.o dune3d.p/src_document_entity_entity_line3d.cpp.o dune3d.p/src_document_entity_entity_line2d.cpp.o dune3d.p/src_document_entity_entity_arc2d.cpp.o dune3d.p/src_document_entity_entity_circle2d.cpp.o dune3d.p/src_document_entity_entity_circle3d.cpp.o dune3d.p/src_document_entity_entity_arc3d.cpp.o dune3d.p/src_document_entity_entity_workplane.cpp.o dune3d.p/src_document_entity_entity_step.cpp.o dune3d.p/src_document_entity_entity_point2d.cpp.o dune3d.p/src_document_constraint_constraint.cpp.o dune3d.p/src_document_constraint_constraint_points_coincident.cpp.o dune3d.p/src_document_constraint_constraint_parallel.cpp.o dune3d.p/src_document_constraint_constraint_equal_length.cpp.o dune3d.p/src_document_constraint_constraint_equal_radius.cpp.o dune3d.p/src_document_constraint_constraint_point_on_line.cpp.o dune3d.p/src_document_constraint_constraint_point_on_line_base.cpp.o dune3d.p/src_document_constraint_constraint_point_on_circle.cpp.o dune3d.p/src_document_constraint_constraint_same_orientation.cpp.o dune3d.p/src_document_constraint_constraint_hv.cpp.o dune3d.p/src_document_constraint_constraint_point_distance.cpp.o dune3d.p/src_document_constraint_constraint_point_distance_hv.cpp.o dune3d.p/src_document_constraint_constraint_workplane_normal.cpp.o dune3d.p/src_document_constraint_constraint_midpoint.cpp.o dune3d.p/src_document_constraint_constraint_diameter_radius.cpp.o dune3d.p/src_document_constraint_constraint_arc_line_tangent.cpp.o dune3d.p/src_document_constraint_constraint_arc_arc_tangent.cpp.o dune3d.p/src_document_constraint_constraint_line_points_perpendicular.cpp.o dune3d.p/src_document_constraint_constraint_angle.cpp.o dune3d.p/src_document_constraint_constraint_point_in_plane.cpp.o dune3d.p/src_document_constraint_constraint_point_line_distance.cpp.o dune3d.p/src_document_constraint_constraint_point_plane_distance.cpp.o dune3d.p/src_document_constraint_constraint_lock_rotation.cpp.o dune3d.p/src_document_solid_model.cpp.o dune3d.p/src_document_solid_model_occ.cpp.o dune3d.p/src_document_solid_model_array.cpp.o dune3d.p/src_document_solid_model_local_operation.cpp.o dune3d.p/src_document_solid_model_lathe.cpp.o dune3d.p/src_document_solid_model_extrude.cpp.o dune3d.p/src_document_solid_model_util.cpp.o dune3d.p/src_document_group_group.cpp.o dune3d.p/src_document_group_group_sketch.cpp.o dune3d.p/src_document_group_group_reference.cpp.o dune3d.p/src_document_group_group_extrude.cpp.o dune3d.p/src_document_group_group_local_operation.cpp.o dune3d.p/src_document_group_group_fillet.cpp.o dune3d.p/src_document_group_group_chamfer.cpp.o dune3d.p/src_document_group_group_lathe.cpp.o dune3d.p/src_document_group_group_sweep.cpp.o dune3d.p/src_document_group_group_linear_array.cpp.o dune3d.p/src_document_group_group_polar_array.cpp.o dune3d.p/src_document_group_group_array.cpp.o dune3d.p/src_document_export_paths.cpp.o dune3d.p/src_system_system.cpp.o dune3d.p/src_logger_logger.cpp.o dune3d.p/src_logger_log_dispatcher.cpp.o dune3d.p/src_render_renderer.cpp.o dune3d.p/src_util_util.cpp.o dune3d.p/src_util_fs_util.cpp.o dune3d.p/src_util_json_util.cpp.o dune3d.p/src_util_glm_util.cpp.o dune3d.p/src_util_selection_util.cpp.o dune3d.p/src_util_file_version.cpp.o dune3d.p/src_core_core.cpp.o dune3d.p/src_core_tool.cpp.o dune3d.p/src_core_create_tool.cpp.o dune3d.p/src_core_tools_tool_common.cpp.o dune3d.p/src_core_tools_tool_draw_line_3d.cpp.o dune3d.p/src_core_tools_tool_draw_point_2d.cpp.o dune3d.p/src_core_tools_tool_draw_arc_2d.cpp.o dune3d.p/src_core_tools_tool_draw_circle_2d.cpp.o dune3d.p/src_core_tools_tool_draw_contour.cpp.o dune3d.p/src_core_tools_tool_delete.cpp.o dune3d.p/src_core_tools_tool_move.cpp.o dune3d.p/src_core_tools_tool_constrain_coincident.cpp.o dune3d.p/src_core_tools_tool_constrain_hv.cpp.o dune3d.p/src_core_tools_tool_constrain_distance.cpp.o dune3d.p/src_core_tools_tool_constrain_same_orientation.cpp.o dune3d.p/src_core_tools_tool_constrain_parallel.cpp.o dune3d.p/src_core_tools_tool_constrain_workplane_normal.cpp.o dune3d.p/src_core_tools_tool_constrain_midpoint.cpp.o dune3d.p/src_core_tools_tool_constrain_equal_length.cpp.o dune3d.p/src_core_tools_tool_constrain_equal_radius.cpp.o dune3d.p/src_core_tools_tool_constrain_diameter_radius.cpp.o dune3d.p/src_core_tools_tool_constrain_perpendicular.cpp.o dune3d.p/src_core_tools_tool_constrain_point_in_plane.cpp.o dune3d.p/src_core_tools_tool_constrain_lock_rotation.cpp.o dune3d.p/src_core_tools_tool_enter_datum.cpp.o dune3d.p/src_core_tools_tool_draw_workplane.cpp.o dune3d.p/src_core_tools_tool_add_anchor.cpp.o dune3d.p/src_core_tools_tool_move_anchor.cpp.o dune3d.p/src_core_tools_tool_toggle_construction.cpp.o dune3d.p/src_core_tools_tool_import_step.cpp.o dune3d.p/src_core_tools_tool_select_edges.cpp.o dune3d.p/src_core_tools_tool_set_workplane.cpp.o dune3d.p/src_core_tools_tool_helper_constrain.cpp.o dune3d.p/src_core_tools_tool_rotate.cpp.o dune3d.p/src_core_tools_tool_draw_regular_polygon.cpp.o dune3d.p/src_core_tools_tool_draw_rectangle.cpp.o dune3d.p/src_util_history_manager.cpp.o dune3d.p/src_action_action.cpp.o dune3d.p/src_action_action_catalog.cpp.o dune3d.p/src_in_tool_action_in_tool_action_catalog.cpp.o dune3d.p/src_util_str_util.cpp.o dune3d.p/src_util_lut.cpp.o dune3d.p/src_preferences_preferences.cpp.o dune3d.p/src_util_action_label.cpp.o dune3d.p/src_editor_tool_popover.cpp.o dune3d.p/src_editor_workspace_browser.cpp.o dune3d.p/src_widgets_constraints_box.cpp.o dune3d.p/src_editor_document_view.cpp.o dune3d.p/src_editor_group_editor.cpp.o dune3d.p/src_editor_selection_editor.cpp.o dune3d.p/src_dialogs_dialogs.cpp.o dune3d.p/src_dialogs_enter_datum_window.cpp.o dune3d.p/src_dialogs_rotate_window.cpp.o dune3d.p/src_dialogs_tool_window.cpp.o dune3d.p/src_widgets_spin_button_dim.cpp.o dune3d.p/src_widgets_spin_button_angle.cpp.o dune3d.p/src_util_gtk_util.cpp.o dune3d.p/src_preferences_preferences_window.cpp.o dune3d.p/src_preferences_action_editor.cpp.o dune3d.p/src_preferences_preferences_window_keys.cpp.o dune3d.p/src_preferences_preferences_window_in_tool_keys.cpp.o dune3d.p/src_preferences_preferences_window_keys_base.cpp.o dune3d.p/src_preferences_preferences_window_canvas.cpp.o dune3d.p/src_preferences_preferences_row.cpp.o dune3d.p/src_preferences_preferences_window_misc.cpp.o dune3d.p/src_widgets_capture_dialog.cpp.o dune3d.p/src_widgets_axes_lollipop.cpp.o dune3d.p/src_widgets_about_dialog.cpp.o dune3d.p/src_widgets_recent_item_box.cpp.o dune3d.p/src_widgets_log_view.cpp.o dune3d.p/src_widgets_log_window.cpp.o dune3d.p/src_widgets_clipping_plane_window.cpp.o dune3d.p/src_util_version.cpp.o dune3d.p/src_util_msd.cpp.o dune3d.p/src_util_msd_animator.cpp.o dune3d.p/src_util_exception_util.cpp.o -Wl,-dead_strip_dylibs -Wl,-headerpad_max_install_names -Wl,-rpath,/opt/homebrew/Cellar/gtk4/4.12.4/lib -Wl,-rpath,/opt/homebrew/Cellar/pango/1.50.14/lib -Wl,-rpath,/opt/homebrew/Cellar/harfbuzz/8.3.0/lib -Wl,-rpath,/opt/homebrew/Cellar/gdk-pixbuf/2.42.10_1/lib -Wl,-rpath,/opt/homebrew/Cellar/cairo/1.18.0/lib -Wl,-rpath,/opt/homebrew/Cellar/graphene/1.10.8/lib -Wl,-rpath,/opt/homebrew/Cellar/glib/2.78.2/lib -Wl,-rpath,/opt/homebrew/opt/gettext/lib -Wl,-rpath,/opt/homebrew/Cellar/gtkmm4/4.12.0/lib -Wl,-rpath,/opt/homebrew/Cellar/pangomm/2.50.1/lib -Wl,-rpath,/opt/homebrew/Cellar/glibmm/2.78.0/lib -Wl,-rpath,/opt/homebrew/Cellar/cairomm/1.18.0/lib -Wl,-rpath,/opt/homebrew/Cellar/libsigc++/3.6.0/lib -Wl,-rpath,/opt/homebrew/Cellar/libepoxy/1.5.10/lib 3rd_party/solvespace/libsolvespace.a 3rd_party/Clipper2Lib/libclipper.a -lm /opt/homebrew/Cellar/gtk4/4.12.4/lib/libgtk-4.dylib /opt/homebrew/Cellar/pango/1.50.14/lib/libpangocairo-1.0.dylib /opt/homebrew/Cellar/pango/1.50.14/lib/libpango-1.0.dylib /opt/homebrew/Cellar/harfbuzz/8.3.0/lib/libharfbuzz.dylib /opt/homebrew/Cellar/gdk-pixbuf/2.42.10_1/lib/libgdk_pixbuf-2.0.dylib /opt/homebrew/Cellar/cairo/1.18.0/lib/libcairo-gobject.dylib /opt/homebrew/Cellar/cairo/1.18.0/lib/libcairo.dylib /opt/homebrew/Cellar/graphene/1.10.8/lib/libgraphene-1.0.dylib /opt/homebrew/Cellar/glib/2.78.2/lib/libgio-2.0.dylib /opt/homebrew/Cellar/glib/2.78.2/lib/libgobject-2.0.dylib /opt/homebrew/Cellar/glib/2.78.2/lib/libglib-2.0.dylib /opt/homebrew/opt/gettext/lib/libintl.dylib /opt/homebrew/Cellar/gtkmm4/4.12.0/lib/libgtkmm-4.0.dylib /opt/homebrew/Cellar/pangomm/2.50.1/lib/libpangomm-2.48.dylib /opt/homebrew/Cellar/glibmm/2.78.0/lib/libgiomm-2.68.dylib /opt/homebrew/Cellar/glibmm/2.78.0/lib/libglibmm-2.68.dylib /opt/homebrew/Cellar/cairomm/1.18.0/lib/libcairomm-1.16.dylib /opt/homebrew/Cellar/libsigc++/3.6.0/lib/libsigc-3.0.dylib /opt/homebrew/Cellar/libepoxy/1.5.10/lib/libepoxy.dylib /opt/homebrew/lib/libTKernel.7.7.2.dylib /opt/homebrew/opt/tbb/lib/libtbb.12.10.dylib /opt/homebrew/opt/tbb/lib/libtbbmalloc.2.10.dylib /opt/homebrew/lib/libTKMath.7.7.2.dylib /opt/homebrew/lib/libTKG2d.7.7.2.dylib /opt/homebrew/lib/libTKG3d.7.7.2.dylib /opt/homebrew/lib/libTKGeomBase.7.7.2.dylib /opt/homebrew/lib/libTKBRep.7.7.2.dylib /opt/homebrew/lib/libTKGeomAlgo.7.7.2.dylib /opt/homebrew/lib/libTKTopAlgo.7.7.2.dylib /opt/homebrew/lib/libTKPrim.7.7.2.dylib /opt/homebrew/lib/libTKBO.7.7.2.dylib /opt/homebrew/lib/libTKShHealing.7.7.2.dylib /opt/homebrew/lib/libTKBool.7.7.2.dylib /opt/homebrew/lib/libTKHLR.7.7.2.dylib /opt/homebrew/lib/libTKFillet.7.7.2.dylib /opt/homebrew/lib/libTKOffset.7.7.2.dylib /opt/homebrew/lib/libTKFeat.7.7.2.dylib /opt/homebrew/lib/libTKMesh.7.7.2.dylib /opt/homebrew/lib/libTKXMesh.7.7.2.dylib /opt/homebrew/lib/libTKService.7.7.2.dylib /opt/homebrew/opt/freeimage/lib/libfreeimage.dylib /opt/homebrew/opt/freetype/lib/libfreetype.dylib /opt/homebrew/lib/libTKV3d.7.7.2.dylib /opt/homebrew/lib/libTKOpenGl.7.7.2.dylib /opt/homebrew/lib/libTKMeshVS.7.7.2.dylib /opt/homebrew/lib/libTKCDF.7.7.2.dylib /opt/homebrew/lib/libTKLCAF.7.7.2.dylib /opt/homebrew/lib/libTKCAF.7.7.2.dylib /opt/homebrew/lib/libTKBinL.7.7.2.dylib /opt/homebrew/lib/libTKXmlL.7.7.2.dylib /opt/homebrew/lib/libTKBin.7.7.2.dylib /opt/homebrew/lib/libTKXml.7.7.2.dylib /opt/homebrew/lib/libTKStdL.7.7.2.dylib /opt/homebrew/lib/libTKStd.7.7.2.dylib /opt/homebrew/lib/libTKTObj.7.7.2.dylib /opt/homebrew/lib/libTKBinTObj.7.7.2.dylib /opt/homebrew/lib/libTKXmlTObj.7.7.2.dylib /opt/homebrew/lib/libTKVCAF.7.7.2.dylib /opt/homebrew/lib/libTKXDE.7.7.2.dylib /opt/homebrew/lib/libTKXSBase.7.7.2.dylib /opt/homebrew/lib/libTKSTEPBase.7.7.2.dylib /opt/homebrew/lib/libTKSTEPAttr.7.7.2.dylib /opt/homebrew/lib/libTKSTEP209.7.7.2.dylib /opt/homebrew/lib/libTKSTEP.7.7.2.dylib /opt/homebrew/lib/libTKIGES.7.7.2.dylib /opt/homebrew/lib/libTKXCAF.7.7.2.dylib /opt/homebrew/lib/libTKXDEIGES.7.7.2.dylib /opt/homebrew/lib/libTKXDESTEP.7.7.2.dylib /opt/homebrew/lib/libTKSTL.7.7.2.dylib /opt/homebrew/lib/libTKRWMesh.7.7.2.dylib /opt/homebrew/lib/libTKVRML.7.7.2.dylib /opt/homebrew/lib/libTKXmlXCAF.7.7.2.dylib /opt/homebrew/lib/libTKBinXCAF.7.7.2.dylib /opt/homebrew/lib/libTKXDECascade.7.7.2.dylib /opt/homebrew/lib/libTKDraw.7.7.2.dylib /opt/homebrew/opt/tcl-tk/lib/libtcl8.6.dylib /opt/homebrew/opt/tcl-tk/lib/libtk8.6.dylib /opt/homebrew/lib/libTKTopTest.7.7.2.dylib /opt/homebrew/lib/libTKOpenGlTest.7.7.2.dylib /opt/homebrew/lib/libTKViewerTest.7.7.2.dylib /opt/homebrew/lib/libTKDCAF.7.7.2.dylib /opt/homebrew/lib/libTKXSDRAW.7.7.2.dylib /opt/homebrew/lib/libTKXDEDRAW.7.7.2.dylib /opt/homebrew/lib/libTKTObjDRAW.7.7.2.dylib /opt/homebrew/lib/libTKQADraw.7.7.2.dylib /opt/homebrew/lib/libTKExpress.7.7.2.dylib -lobjc -framework OpenGL -framework AppKit -framework IOKit

./dune3d
```

That big command is there because the one that meson generates is incorrect (see linked issue in my last comment).

**Comment by vespakoen** [comment]
@tkornack I tried to run a build in github actions but ran into another issue where pygobject3 is not being found somehow: https://github.com/vespakoen/dune3d/actions/runs/7409168550/job/20158883349 but most other dependencies seem to be there, and I think the issue might be coming from the python in the github runner? anyways, hope you can get it to run ;)

File: src/render/renderer.cpp [hunk 1]
```diff
@@ -413,7 +413,9 @@ void Renderer::draw_distance_line(const glm::vec3 &from, const glm::vec3 &to, co
     m_ca.add_selectable(m_ca.draw_line(from, p1), sr);
     m_ca.add_selectable(m_ca.draw_line(to, p2), sr);
 
-    std::string label = std::format(" {:.3f}", distance);
+    std::stringstream ss;
```
Comment (carrotIndustries) [inline src/render/renderer.cpp:416]: I'd very much prefer not having to use `stringstream` anymore. Any other compiler supports `format` by now. Is there no more modern GCC or LLVM in homebrew?

Comment (vespakoen) [reply src/render/renderer.cpp:416]: There is indeed, llvm and gcc are both in homebrew, but getting them to compile the project didn't work for me yet, here is what I tried so far:


macos.ini
```
[binaries]
; c = 'gcc-13'
; cpp = 'g++-13'
c = '/opt/homebrew/opt/llvm/bin/clang'
cpp = '/opt/homebrew/opt/llvm/bin/clang++'
ar = '/opt/homebrew/opt/llvm/bin/llvm-ar'
pkg-config = 'pkg-config'
cmake = 'cmake'

[built-in options]
cpp_args = [ '-arch', 'arm64', '-I', '/opt/homebrew/opt/llvm/include', '-isystem', '/usr/include', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk', '-mmacosx-version-min=11.0' ]
cpp_link_args = [ '-arch', 'arm64', '-L/opt/homebrew/opt/llvm/lib', '-isystem', '/usr/include', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk', '-Wl,-ld_classic' ]
c_args = [ '-arch', 'arm64', '-I', '/opt/homebrew/opt/llvm/include', '-isystem', '/usr/include', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk', '-mmacosx-version-min=11.0' ]
c_link_args = [ '-arch', 'arm64', '-L/opt/homebrew/opt/llvm/lib', '-isystem', '/usr/include', '-isysroot', '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk', '-Wl,-ld_classic' ]

[host_machine]
system = 'darwin'
cpu_family = 'aarch64'
cpu = 'arm64'
endian = 'little'
```

And compile with:

```
rm -rf build && meson setup --cross-file macos.ini build
meson compile -C build
```

I'll check again later what kind of issues I ran into with this...

Comment (vespakoen) [reply src/render/renderer.cpp:416]: Just ran it again and it seems to compile now, not sure what happened in between on my computer, or maybe I was thrown off by the meson issue before. I opened a new PR that uses llvm from homebrew with the cross compile configuration.

Comment (tkornack) [reply src/render/renderer.cpp:416]: > I'd very much prefer not having to use `stringstream` anymore. Any other compiler supports `format` by now. Is there no more modern GCC or LLVM in homebrew?

We could try enabling?
https://stackoverflow.com/questions/71777564/cant-use-stdformat-in-c20

**Comment by tkornack** [comment]
This worked, after adding the following pieces:

brew install libffi zlib expat ossp-uuid librsvg

PKG_CONFIG_PATH="/opt/homebrew/opt/libffi/lib/pkgconfig:/opt/homebrew/opt/zlib/lib/pkgconfig:/opt/homebrew/opt/expat/lib/pkgconfig"

and in the build directory:

**Comment by vespakoen** [comment]
Ahhh I think I also disabled / removed some things in the solvespace submodule which are not needed for dune3d, will take a look later and report back.

I am wondering though, if you also have the issue at the linking step, and if not, what version of meson you have :)

**Comment by tkornack** [comment]
Running now! I see two-finger zoom working, but how do you rotate the view?

**Comment by vespakoen** [comment]
right-mouse drag, with my trackpad I press both fingers down and drag around.

**Comment by vespakoen** [comment]
Closing this in favor of #45

**Pull request closed without merging**